src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1202,43 +1156,45 @@ public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Insta
      *     Org namespace to search
      * @return Consumer that matches the given
      */
-    @SuppressWarnings("unchecked")
-    @Transactional
-    public CandlepinQuery<Consumer> getHypervisorsBulk(Iterable<String> hypervisorIds, String ownerId) {
+    public List<Consumer> getHypervisorsBulk(Collection<String> hypervisorIds, String ownerId) {
         if (hypervisorIds == null || !hypervisorIds.iterator().hasNext()) {
-            return this.cpQueryFactory.<Consumer>buildQuery();
+            return new ArrayList<>();
         }
 
-        DetachedCriteria criteria = DetachedCriteria.forClass(Consumer.class)
-            .createAlias("hypervisorId", "hvsr")
-            .add(Restrictions.eq("ownerId", ownerId))
-            .add(this.getHypervisorIdRestriction(hypervisorIds))
-            .addOrder(org.hibernate.criterion.Order.asc("hvsr.hypervisorId"))
-            .setFetchMode("type", FetchMode.SELECT);
+        List<String> lowerCaseHypervisorIds = hypervisorIds.stream()
+            .map(String::toLowerCase)
+            .toList();
 
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria)
-            .setLockMode(LockModeType.PESSIMISTIC_WRITE);
-    }
+        String jpql = """
+            SELECT c FROM Consumer c
+            JOIN c.hypervisorId h
+            WHERE c.ownerId = :ownerId
+                AND LOWER(h.hypervisorId) IN :lowerCaseHypervisorIds
+            ORDER BY h.hypervisorId ASC""";
+
+        List<Consumer> consumers = new ArrayList<>();
```

Comment:
This addresses the partitioning, but now we're at risk of three other problems:
- The lock mode on the query has gone missing, so any locking behavior existing callers may be expecting is gone.
- The input may or may not have duplicate entries, and if those duplicates exist in different partitions, we're guaranteed to get duplicate outputs since we lose the implied uniqueness from the schema.
- If the locking is restored, we're at risk of deadlock depending on the order of our input since we don't know which partition an input will end up in; we could end up locking consumers A, B, and C in various orders. Two or more threads doing this with overlapping partitioned inputs will eventually deadlock.

Reply:
done